static/javascript/launch.js

```diff
@@ -152,54 +165,72 @@ function clearSurveyMetadataFields() {
   showMetadataAccordion("sds", false);
 }
 
-function setSurveyType(event) {
-  localStorage.setItem("survey_type", event.value);
-  setLaunchType("remote");
+function toggleLoadMetadataButton() {
+  if (surveyType && (cirSchema || schemaUrl)) {
+    enableButtons([loadMetadataButton]);
+  } else {
+    disableButtons([loadMetadataButton]);
+  }
 }
 
-function setCirSchema(event) {
-  localStorage.setItem("cir_schema", event.value);
-  setLaunchType("cir");
+function setSurveyType(event) {
+  surveyType = remoteSchemaSurveyType.value;
+  localStorage.setItem("survey_type", surveyType);
+  setLaunchType("remote");
+  toggleLoadMetadataButton();
 }
 
 function setSchemaUrl(event) {
-  localStorage.setItem("schema_url", event.value);
+  schemaUrl = document.querySelector("#remote-schema-url").value;
+  localStorage.setItem("schema_url", schemaUrl);
   setLaunchType("url");
+  toggleLoadMetadataButton();
+}
+
+function setCirSchema(event) {
+  cirSchema = document.querySelector("#cir-schemas").value;
+  localStorage.setItem("cir_schema", cirSchema);
+  setLaunchType("cir");
+  toggleLoadMetadataButton();
 }
 
 function setLaunchType(launchType) {
   const schemaName = document.querySelector("#schema_name");
-  const schemaUrl = document.querySelector("#schema-url");
+  const remoteSchemaUrl = document.querySelector("#remote-schema-url");
   const cirSchemas = document.querySelector("#cir-schemas");
   console.log(schemaName);
-  const remoteSchemaSurveyType = document.querySelector(
-    "#remote-schema-survey-type",
-  );
 
   if (["cir", "remote", "url"].includes(launchType)) {
     if (schemaName.selectedIndex) {
       clearSurveyMetadataFields();
-      enableSubmitFlushButtons(false);
+      disableButtons(launchFlushButtons);
```

Comment:
Could we, for clarity, make it an array and:
```js
disableButtons([launchButton, flushButton]);
```
and in the other place:
```js
const launchButton = document.querySelector("#launch-btn");
const flushButton =  document.querySelector("#flush-btn");
];
```
?
Otherwise it might confuse someone into thinking it's a verb and we are trying to launch flush buttons.

Reply:
Otherwise the changes look fine.